### PR TITLE
update openshift deps and ignore undesired version in dependabot

### PR DESCRIPTION
**Pull request by rmfitzpatrick**
Need to ignore [v3.9.0](https://pkg.go.dev/github.com/openshift/api?tab=versions). Using a replace rule in the main go.mod since contrib has similarly accepted dependabot udpate.

### CODE REVIEW

**Review by atoulme — APPROVED** [review]

**Review by jvoravong — APPROVED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -60,13 +62,17 @@ updates:
...
+        versions: ["v3.9.0+incompatible"]
     schedule:
       interval: "weekly"
   - package-ecosystem: "gomod"
     directory: "/pkg/signalfx-agent"
     ignore:
       - dependency-name: "go.opentelemetry.io/*"
       - dependency-name: "github.com/open-telemetry/*"
+      - dependency-name: "github.com/openshift/*"
+        versions: ["v3.9.0+incompatible"]
```
Comment (jeffreyc-splunk) [inline .github/dependabot.yml:75]: Should these changes be added to the 'gendependabot' target in the Makefile instead?

Comment (rmfitzpatrick) [reply .github/dependabot.yml:75]: oh right, forgot about that...

Comment (jeffreyc-splunk) [reply .github/dependabot.yml:75]: Shall I add a simple github job to fail if this file is manually changed?

Comment (rmfitzpatrick) [reply .github/dependabot.yml:75]: Updated but I'm not sure how platform agnostic the changes are. imo we should adopt a templated approach in future changes.

**Pull request merged**